s2/encode_test.go

```diff
@@ -131,7 +145,7 @@ func TestEncoderRegression(t *testing.T) {
 				t.Error(err)
 				return
 			}
-			test(t, b)
+			test(t, b[:len(b):len(b)])
```

Comment:
Might it be a good idea to restrict the capacity of src in EncodeBetter, too? (And maybe other places?)

Reply:
@greatroar That is exactly what this is doing. The `test` function tests both Encode and EncodeBetter using the data.

Reply:
I mean in the actual code, not in the test.